mlkem/indcpa.c

```diff
@@ -46,8 +56,18 @@ static void pack_pk(uint8_t r[MLKEM_INDCPA_PUBLICKEYBYTES], polyvec *pk,
  *              - const uint8_t *packedpk: pointer to input serialized public
  *key
  **************************************************/
-static void unpack_pk(polyvec *pk, uint8_t seed[MLKEM_SYMBYTES],
-                      const uint8_t packedpk[MLKEM_INDCPA_PUBLICKEYBYTES]) {
+CBMC_STATIC
+void unpack_pk(
+    polyvec *pk, uint8_t seed[MLKEM_SYMBYTES],
+    const uint8_t packedpk[MLKEM_INDCPA_PUBLICKEYBYTES])  // clang-format off
+REQUIRES(IS_FRESH(packedpk, MLKEM_INDCPA_PUBLICKEYBYTES))
+REQUIRES(IS_FRESH(pk, sizeof(polyvec)))
+REQUIRES(IS_FRESH(seed, MLKEM_SYMBYTES))
+ENSURES(FORALL(int, k0, 0, MLKEM_K - 1,
+  ARRAY_IN_BOUNDS(int, k1, 0, MLKEM_N - 1, pk->vec[k0].coeffs, 0, MLKEM_Q - 1)))
```

Comment:
same.

Reply:
fixed

Reply:
We need to revisit this at some point - but it's fine for now.